### PR TITLE
CB-16451: Fix issue when getting SDX events for environment with deleted datalakes.

### PR DESCRIPTION
**Pull request by sxxgrc**
Jira: https://jira.cloudera.com/browse/CB-16451

Please look at the Jira description for the reason behind this bug and how I resolved it. Essentially, SDX events will run into an error if any of the environment's DLs are deleted, this being a result of additional error handling we added into 2.54.0.

This simply does nothing for the cloudbreak events if the DL is deleted, allowing for its SDX events to still be present, but not breaking the overall event retrieval.

I have tested this both via local UI and through both endpoint methods via Postman.

### CODE REVIEW

**Comment by sxxgrc** [comment]
I want this to go into 2.54.0 since the bug was introduced in that branch. I don't expect many people to run into issues, I believe it mainly only happens with processes such as resize, but I want to make sure the branch is perfectly safe.

**Review by horadla23 — DISMISSED** [review]
so we will get events from datalake service but not from cloudbreak in case of deleted/detached datalake?

**Comment by horadla23** [comment]
pls rebase your PR onto top of CB-2.54.0

**Comment by sxxgrc** [comment]
> so we will get events from datalake service but not from cloudbreak in case of deleted/detached datalake?

Yes exactly, which is desirable, for example in the resize case, where the events from the original deleted datalake contain useful info about the resize process overall.

File: datalake/src/main/java/com/sequenceiq/datalake/service/SdxEventsService.java [hunk 1]
```diff
@@ -117,6 +118,10 @@
     }
 
     private List<CDPStructuredEvent> retrievePagedCloudbreakServiceEvents(SdxCluster sdxCluster, Integer page, Integer size) {
+        if (sdxCluster.getDeleted() != null) {
+            return Collections.emptyList();
+        }
+
```
Comment (kkalvagadda1) [inline datalake/src/main/java/com/sequenceiq/datalake/service/SdxEventsService.java:124]: We are intentionally getting the events for deleted datalake as well. When the datalake is resized, we also need to show the events generated for the data lake deleted.

What is the motivation for this change?

**Review by kkalvagadda1 — CHANGES_REQUESTED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/service/SdxEventsService.java (hunk 1 above)
Comment (sxxgrc) [reply datalake/src/main/java/com/sequenceiq/datalake/service/SdxEventsService.java:124]: Getting Cloudbreak events from deleted datalakes literally does not work. This will throw an error and show nothing on the UI. Events from the SDX service are still shown.
I tested this specifically with a resize case and made sure the correct events are shown.

**Comment by horadla23** [comment]
I will dismiss my review and let you (as feature creator/owner) decide what is the best solution here
